docs/user-guide/replicaset.md

```diff
@@ -0,0 +1,45 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Replica Set
+
+## What is a _replica set_?
+
+Replica set is the next generation replication controller. The only difference between a _replica set_ and a [_replication controller_](replication-controller.md) right now is the selector support. Replica set supports the new set-based selector requirements as described in the [labels user guide](labels.md#label-selectors) whereas a replication controller only supports equality-based selector requirements.
+
+While replica sets can be used indepently, today it's mainly used by [deployments](deployments.md) as a mechanism to orchestrate pod creation, deletion and updates.
```

Comment:
> is it possible to describe in more detail, how it can be used by deployments? 

Expanded the deployments paragraph a bit to clarify this.

> also it would help users, to note any differences while using ReplicaSet vs ReplicationController with hpa

@r8k Added a section and an example.

> Also need use case for when you'd want to use RS and when to use Deployment.

@janetkuo Done. Added "When to use a replica set?" section.

> @r8k I'm not sure @madhusudancs can explain more regarding "how [replica sets] can be used by deployments". The point being made was that deployments use replica sets behind the scenes to manage the desired state described in a deployment. I could be wrong but that's how I understand it.

@whitlockjc You are exactly right. Just added that clarification to the doc, but worded it in a slightly different way. If you think you can improve it, please feel free to send a PR after this is merged.